File: pec-observation-app/ios-app/Sources/PECObservation/Domain/Entities/Feedback.swift
import Foundation

public enum FeedbackStatus: String, Codable, Equatable {
    case draft = "DRAFT"
    case inReview = "IN_REVIEW"
    case humanReviewed = "HUMAN_REVIEWED"
    case approved = "APPROVED"

    public var displayName: String {
        switch self {
        case .draft: return "Rascunho IA"
        case .inReview: return "Em Revisão"
        case .humanReviewed: return "Revisado"
        case .approved: return "Aprovado"
        }
    }
}

public struct FeedbackItem: Codable, Equatable {
    public var title: String
    public var description: String
    public var evidence: String?

    public init(title: String, description: String, evidence: String? = nil) {
        self.title = title
        self.description = description
        self.evidence = evidence
    }
}

public struct ActionItem: Identifiable, Codable, Equatable {
    public let id: UUID
    public var description: String
    public var owner: String
    public var dueDate: String?
    public var expectedEvidence: String?
    public var isCompleted: Bool

    public init(id: UUID = UUID(), description: String, owner: String = "Professor",
                dueDate: String? = nil, expectedEvidence: String? = nil,
                isCompleted: Bool = false) {
        self.id = id
        self.description = description
        self.owner = owner
        self.dueDate = dueDate
        self.expectedEvidence = expectedEvidence
        self.isCompleted = isCompleted
    }
}

public struct Feedback: Identifiable, Codable, Equatable {
    public let id: UUID
    public let observationId: UUID
    public var summary: String?
    public var strengths: [FeedbackItem]
    public var improvementPoints: [FeedbackItem]
    public var actionItems: [ActionItem]
    public var status: FeedbackStatus
    public var version: Int
    public var reviewedAt: Date?

    public init(id: UUID = UUID(), observationId: UUID, summary: String? = nil,
                strengths: [FeedbackItem] = [], improvementPoints: [FeedbackItem] = [],
                actionItems: [ActionItem] = [], status: FeedbackStatus = .draft,
                version: Int = 1, reviewedAt: Date? = nil) {
        self.id = id
        self.observationId = observationId
        self.summary = summary
        self.strengths = strengths
        self.improvementPoints = improvementPoints
        self.actionItems = actionItems
        self.status = status
        self.version = version
        self.reviewedAt = reviewedAt
    }
}

File: pec-observation-app/services/ms-007-feedback/app/models/feedback.py
from datetime import datetime
from sqlalchemy import Column, DateTime, Integer, JSON, String, Text
from pec_shared.models_base import Base, gen_uuid


class Feedback(Base):
    __tablename__ = "feedbacks"

    id = Column(String, primary_key=True, default=gen_uuid)
    observation_id = Column(String, nullable=False, unique=True, index=True)
    strengths = Column(JSON, nullable=True)
    improvement_points = Column(JSON, nullable=True)
    evidence = Column(JSON, nullable=True)
    action_plan = Column(JSON, nullable=True)
    summary = Column(Text, nullable=True)
    status = Column(String, nullable=False, default="DRAFT")
    version = Column(Integer, nullable=False, default=1)
    reviewed_by = Column(String, nullable=True)
    reviewed_at = Column(DateTime, nullable=True)
    approved_by = Column(String, nullable=True)
    approved_at = Column(DateTime, nullable=True)
    created_at = Column(DateTime, default=datetime.utcnow)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)


class FeedbackVersion(Base):
    __tablename__ = "feedback_versions"

    id = Column(String, primary_key=True, default=gen_uuid)
    feedback_id = Column(String, nullable=False, index=True)
    version_number = Column(Integer, nullable=False)
    snapshot = Column(JSON, nullable=True)  # full feedback state
    edited_by = Column(String, nullable=True)
    edit_reason = Column(String, nullable=True)
    created_at = Column(DateTime, default=datetime.utcnow)


class ActionItem(Base):
    __tablename__ = "action_items"

    id = Column(String, primary_key=True, default=gen_uuid)
    feedback_id = Column(String, nullable=False, index=True)
    observation_id = Column(String, nullable=False)
    description = Column(Text, nullable=False)
    owner = Column(String, nullable=True)  # Professor | PEC | Coordenação
    due_date = Column(String, nullable=True)
    expected_evidence = Column(Text, nullable=True)
    is_completed = Column(String, nullable=False, default="false")
    completed_at = Column(DateTime, nullable=True)
    created_at = Column(DateTime, default=datetime.utcnow)
